jimp.d.ts

```diff
@@ -234,369 +234,266 @@ declare namespace Jimp {
-            w: number,
-            h: number
+            cb?: Jimp.ImageCallback<T>
         ): this;
-        pixelate<T>(
+        opaque(cb?: Jimp.ImageCallback<T>): this;
+        pixelate(size: number, cb?: Jimp.ImageCallback<T>): this;
+        pixelate(
             size: number,
             x: number,
             y: number,
             w: number,
             h: number,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        convolute<T>(kernel: number[][], cb: Jimp.ImageCallback<T>): T;
-        convolute(
-            kernel: number[][],
-            x: number,
-            y: number,
-            w: number,
-            h: number
+            cb?: Jimp.ImageCallback<T>
         ): this;
-        convolute<T>(
+        convolute(kernel: number[][], cb?: Jimp.ImageCallback<T>): this;
+        convolute(
             kernel: number[][],
             x: number,
             y: number,
             w: number,
             h: number,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        color(actions: ColorAction[]): this;
-        color<T>(actions: ColorAction[], cb: Jimp.ImageCallback<T>): T;
-        colour(actions: ColorAction[]): this;
-        colour<T>(actions: ColorAction[], cb: Jimp.ImageCallback<T>): T;
+            cb?: Jimp.ImageCallback<T>
+        ): this;
+        color(actions: ColorAction[], cb?: Jimp.ImageCallback<T>): this;
+        colour(actions: ColorAction[], cb?: Jimp.ImageCallback<T>): this;
 
         // Shape methods
-        rotate<T>(deg: number, cb: Jimp.ImageCallback<T>): T;
-        rotate(deg: number, mode?: number | boolean): this;
-        rotate<T>(
+        rotate(deg: number, cb?: Jimp.ImageCallback<T>): this;
+        rotate(
             deg: number,
             mode: number | boolean | undefined | null,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        flip(horizontal: boolean, vertical: boolean): this;
-        flip<T>(
+            cb?: Jimp.ImageCallback<T>
+        ): this;
+        flip(
             horizontal: boolean,
             vertical: boolean,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        mirror(horizontal: boolean, vertical: boolean): this;
-        mirror<T>(
+            cb?: Jimp.ImageCallback<T>
+        ): this;
+        mirror(
             horizontal: boolean,
             vertical: boolean,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        resize<T>(w: number, h: number, cb: Jimp.ImageCallback<T>): T;
-        resize(w: number, h: number, mode?: string): this;
-        resize<T>(
+            cb?: Jimp.ImageCallback<T>
+        ): this;
+        resize(w: number, h: number, cb?: Jimp.ImageCallback<T>): this;
+        resize(
             w: number,
             h: number,
             mode: string | undefined | null,
-            cb: Jimp.ImageCallback<T>
-        ): T;
-        cover<T>(w: number, h: number, cb: Jimp.ImageCallback<T>): T;
-        cover<T>(
+            cb?: Jimp.ImageCallback<T>

```

Comment:
Since `cb` is optional here, you can make `mode` optional instead of `string | undefined | null`. The same goes for all the other ` | undefined | null` that are now followed by an optional callback